File: Unidad 9/Ejercicio5/SignalR/TresEnRayaSignalR/TresEnRayaSignalR/Entities/Movimiento.cs
namespace TresEnRayaSignalR.Entities
{
    /// <summary>
    /// Representa un movimiento realizado por un jugador en la partida.
    /// Contiene la posición del tablero y el símbolo del jugador.
    /// </summary>
    public class Movimiento
    {
        /// <summary>
        /// Posición del movimiento en el tablero (fila y columna).
        /// </summary>
        public int[] Posicion { get; set; } = new int[2];

        /// <summary>
        /// Símbolo del jugador que realiza el movimiento ("X" u "O").
        /// </summary>
        public string Simbolo { get; set; } = string.Empty;

        /// <summary>
        /// Constructor que inicializa la posición y el símbolo del movimiento.
        /// </summary>
        /// <param name="posicion">Coordenadas del tablero (fila y columna).</param>
        /// <param name="simbolo">Símbolo del jugador.</param>
        public Movimiento(int[] posicion, string simbolo)
        {
            Posicion = posicion;
            Simbolo = simbolo;
        }

        /// <summary>
        /// Constructor vacío.
        /// </summary>
        public Movimiento() { }
    }
}

File: Unidad 9/Ejercicio5/React Native/TresEnRayaRN/src/domain/entities/Movimiento.ts
/**
 * Representa un movimiento en el tablero del Tres en Raya.
 * Contiene la posición (fila y columna) y el símbolo del jugador.
 */
export class Movimiento {
  /** Posición del movimiento en el tablero [fila, columna] */
  posicion: number[];

  /** Símbolo del jugador que realiza el movimiento ("X" u "O") */
  simbolo: string;

  /**
   * Crea un nuevo movimiento con la posición y el símbolo especificados.
   * @param {number[]} posicion - Coordenadas del tablero [fila, columna].
   * @param {string} simbolo - Símbolo del jugador que realiza el movimiento.
   */
  constructor(posicion: number[], simbolo: string) {
    this.posicion = posicion;
    this.simbolo = simbolo;
  }
}
